engine/src/main/java/org/pentaho/di/www/JobMap.java

```diff
@@ -128,6 +129,21 @@ public synchronized CarteObjectEntry getFirstCarteObjectEntry( String jobName )
     return null;
   }
 
+  public synchronized CarteObjectEntry getUniqueCarteObjectEntry( String jobName ) throws DuplicateKeyException {
```

Comment:
Can I get a few unit tests for this method?
1. `jobName` with null value
2. `jobName` with empty value
3. When only one job with the given name exists
4. When multiple jobs with the given name exist